File: lecture_002/fibonacci.java
//  fibonacci series

import java.util.Scanner;

public class fibonacci{
    public static void main(String[] args){
        Scanner scn=new Scanner(System.in);
        int n, next = 1, prev = 0, sum = 0;
        System.out.print("enter the n no. : ");
        n = scn.nextInt();

        if (n >= 1){
            System.out.print(prev + " ");
            if (n == 1){
                scn.close();                //  used to close Scanner util after running it
                return;
            }
        }
        
        if (n >= 2){
            System.out.print(next + " ");
            if (n == 2){
                scn.close();
                return;
            }
        }

        for (int i=3; i <= n; i++){
            sum = prev + next;
            prev = next;
            next = sum;
            System.out.print(sum + " ");
        }
        System.out.println();
        System.out.print("Nth digit is : \n" + sum);

        scn.close();
    }
}

File: lecture_002/fibonacci.cpp
//  fibonacci series

#include<iostream>

using namespace std;
int main(int args, char**argv){
	int n, next = 1, prev = 0, sum = 0;
	cout<<"enter the n no. : ";
	cin>>n;

	if (n >= 1)
	{
		cout<<prev<<" ";
		if (n == 1)
		{
			return 0;
		}
	}

	if (n >= 2)
	{
		cout<<next<<" ";
		if (n == 2)
		{
			return 0;
		}
	}

	for (int i = 3; i <= n; i++)
	{
		sum = prev + next;
		prev = next;
		next = sum;
		cout<<sum<<" ";
	}
	cout<<endl;
	cout<<"Nth digit is : \n"<<sum<<endl;
	return 0;
}